### ISSUE TITLE
Invalid mount options on service should produce error

### ISSUE THREAD

**Issue by thaJeztah** [docker/docker#24818]
Trying to create a service with invalid mount options should produce an error instead of creating the service (and keep trying);

For example:

    docker service create --name foo --mount type=volume,target=/ nginx:alpine

Is invalid, because `/` cannot be used as a target. This error will never be solvable, so there's no reason to create the service (without producing an error)

similar to https://github.com/docker/docker/issues/24134

**Comment by stevvooe** [comment]
@thaJeztah Unfortunately, the meaning of the path is specific to the targeted platform. These can't be validated until assigned to a specific node/platform.

In general, we need to work on the error reporting. Let's revisit this after the release and see where we land.

**Comment by thaJeztah** [comment]
@stevvooe SGTM; I'll keep it on the milestone, then after release we can decide to move it to 1.13, or possibly a 1.12.1 with improvements

**Comment by cpuguy83** [comment]
@stevvooe The path should be able to be evaluated.
`/` is not allowed as a container path, and neither is `c:\`, these are pretty well-defined rules that won't change.

**Comment by cpuguy83** [comment]
At least somewhere to start, anyway.

**Comment by thaJeztah** [comment]
@cpuguy83 good point, I thought about that later; think we should validate that? (There's still other situations of course, but it's a start)
